### PR TITLE
[kube-prometheus-stack] Add missing fields to prometheus and alertmanager

### PR DESCRIPTION
**Pull request by bashofmann**
#### What this PR does / why we need it:
I went through the api specs of prometheus-operator 0.42.1 and added all the missing fields to the Prometheus and Alertmanager resources that I could find.

See https://github.com/prometheus-operator/prometheus-operator/blob/v0.42.1/Documentation/api.md

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by bashofmann** [comment]
Rebased and changed it to a minor version bump.

**Comment by bashofmann** [comment]
Rebased. @monotek is there something else that blocks merging this?

**Review by monotek — CHANGES_REQUESTED** [review]
Please update version again.

I cant merge your pr, as I'm not one of the maintainers of the chart.

**Review by monotek — APPROVED** [review]

**Review by gkarthiks — APPROVED** [review]

**Pull request merged**
